documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,417 @@
+           -import -file ca-cert -storepass changeit
+
+   # Import the signed server certificate into the keystore.
+   $ keytool -keystore controller01.server.jks -alias controller01 -noprompt \
+           -import -file controller01.pem -storepass changeit
+   ```
+
+5. Export each server's key into a separate `PEM` file.
+
+   This is a two step process.
+   * First, convert the server's keystore in `.jks` format into `.p12` format.
+
+     ```bash
+     $ keytool -importkeystore
+             -srckeystore controller01.jks \
+             -destkeystore controller01.p12 \
+             -srcstoretype jks -deststoretype pkcs12 \
+             -srcstorepass changeit -deststorepass changeit
+     ```
+   * Then, export the private key of the server into a `PEM` file. Note that the generated `PEM` file is not protected
+   by a password. The key itself is password-protected, as we are using the `-nodes` flag. So, be sure
+   to protect it using operating system's technical controls as well as procedural controls.
+
+     ```bash
+     $ openssl pkcs12 -in controller01.p12 -out controller01.key.pem -passin pass:1111_aaaa -nodes
+     ```
+
+Step 5 concludes this stage, and the stage is now set for installing the certificates and other PKI material in Pravega.
+
+The table below lists the key output of this stage. Note that you'll typically need one of each file per Pravega
+service, but you may share the same file for services collocated on the same host for logistical or economical reasons.
+
+| Files | Example File| Description |
+|:-----:| :---:|:--------|
+| Certificate in `PEM` format | `controller01.pem` file | The signed certificate to be used by the service.|
+| Private key in `PEM` format | `controller01.key.pem` file  |  The private key to be used by the service. |
+| Server keystore in `JKS` format | `controller01.server.jks` file | The keystore file to be used by the service. |
+
+
+### Stage 3: Enabling TLS and Deploying Certificates
+
+We'll discuss this in the [next](#enabling-tls-and-auth-in-pravega) section, together with other security configuration and setup.
+
+## Enabling TLS and Auth in Pravega
+
+Enabling TLS and auth in Pravega involves the following steps:
+
+1. Configuring TLS and auth on server side
+2. Configuring TLS and credentials on client Side
+3. Having TLS and auth take effect
+
+These steps are discussed in the following sub-sections.
+
+### Configuring TLS and Auth on Server Side
+
+This step is about using the certificates, keys, keystores and truststores generated earlier to configure
+TLS and Auth for the services on the server side.
+
+Note that if you enable TLS and Auth on one service, you must enable them for all the other services too. Pravega
+does not support using TLS and Auth for only some instances of the services.
+
+You can configure the following services for TLS and Auth:
+
+1. Controller
+2. Segment Store
+3. Zookeeper (optionally)
+4. Bookkeeper (optionally)
```

Comment:
Perhaps we are not ready to talk about that yet, but Tier 2 should be in this list.

Reply:
Yes, I'd like to do try them out and understand their workings before putting them here. We can add them in a later enhancement to this doc.